config/unmarshal.go

```diff
@@ -139,13 +49,18 @@ func unmarshalConfig(r io.Reader, allowEnvironmentOverrides bool) (*model.Config
 		return nil, nil, errors.Wrapf(err, "failed to read")
 	}
 
-	var rawConfig interface{}
+	var rawConfig model.Config
```

Comment:
If we unmarshal directly to `model.Config`, won't we get a `UnmarshalTypeError` if the persisted config happened to contain fields we no longer care about (or contained spurious extra fields)?

Reply:
As an aside, when we deprecate feature flags, do we intend to leave them there permanently? (I imagine yes, given semver requirements.)

Reply:
No. `UnmarshalTypeError` only occurs when the type is wrong. As long as we don't reuse names, this won't bite us. 
I verified the behavior is the same on master, the only difference is that it's the `json` library that reports the error rather than viper.  

For deprecation we remove them from the code, but they live forever in the management system. That way previous MM version can get the correct values.